README.md

```diff
@@ -1,6 +1,185 @@
+
+Example:
+```yaml
+vault:
+  lookup:
+    # These pillar items are templated by the Terraform `template_dir` resource: [https://www.terraform.io/docs/providers/template/r/dir.html]
+    # Input vars can be defined and provided to the module through the `terraform.auto.tfvar` file.
+    # See `variables.var` file for more information on each variables.
+    api_port:  ${api_port}
+  cluster_port:  ${cluster_port}
+  dynamodb_table:  ${dynamodb_table}
+  inbound_cidrs:  ${inbound_cidrs}
+  kms_key_id:  ${kms_key_id}
+  logs_path:  ${logs_path}
+  logs_dir:  ${logs_dir}
+  region:  ${region}
+  ssm_path:  ${ssm_path}
+  version:  ${vault_version}
+
+    secrets_engines:
+      - type:  kv
+        path:  services
+        description:  Sevices specific folders
+        config:
+          default_lease_ttl:  1800
+          max_lease_ttl:  1800
+        secret_config: ${secrets_kv_config}
+      # More secrets engines can be configure here
+    auth_methods:
+      - type:  token
+      path:  token
+      description:  token based credentials
+      config:
+        default_lease_ttl:  0
+        max_lease_ttl:  0
+      auth_config: ${auth_token_config}
+    # More authentication methods can be configure here
+      audit_devices:
+        - type:  file
+          path:  file_log
+          description:  first audit device
+          config:
+          file_path:  /etc/vault/logs/audit.log
+    # More audit devices can be configure here
+    policies:
+      # Following example of vault policy from https://learn.hashicorp.com/vault/identity-access-management/iam-policies
+        - name:  admin
+          content:
+            path:
+              # Manage ad secret engines broadly across Vault
+              'ad/*': {capabilities: [create, read, update, delete, list, sudo]}
+              # List auth methods
+              'sys/auth': {capabilities: [read]}
+              # List existing secret engines.'
+              sys/mounts': {capabilities: [read]}
+              # Read health check
+              'sys/health': {capabilities: [read, sudo]}
+```
+> ***Note***: Additional configurations can be specified for authentication methods using the `auth_config` pillar item. This also applies for secrets engines. Specifying the configuration for a particular secrets engine under the `secret_config` pillar item of that secrets engine type.
+
+In some use cases, passwords or sensitive information will need to be provided in order for Vault to communicate and function properly. For instance, when enabling the [Active Directory Secrets Engine](https://www.vaultproject.io/api/secret/ad/index.html), you need to specify the url of the LDAP server, a bind_dn and a bind_pass to perform user search. This information will need to be hidden from the public and only available to `salt` when it synchronizes the configs. One way to specify these config is via the input  `vault_pillar_extra_config` variable within the `terraform.auto.tfvar` file.
```

Comment:
is this integration limited to LDAPS?

Reply:
This implementation supports all types of secrets engines mentioned [here](https://www.vaultproject.io/api/secret/index.html). This doc is calling out this specific use case of AD. `LDAPS` works as well as `LDAP`.